modules/rostests/kmtests/ntos_cc/CcCopyRead_drv.c

```diff
@@ -315,17 +318,17 @@ TestIrpHandler(
         Status = STATUS_SUCCESS;
     }
 
+// NB: Same block in IoReadWrite_drv.c, MmMapLockedPagesSpecifyCache_drv.c and NtCreateSection_drv.c.
     if (Status == STATUS_PENDING)
     {
         IoMarkIrpPending(Irp);
-        IoCompleteRequest(Irp, IO_NO_INCREMENT);
-        Status = STATUS_PENDING;
+// Q: STATUS_SUCCESS?        Status = STATUS_PENDING; // Added as is in r71445.
```

Comment:
Yeah this wasn't correct in the first place. In the pending case, nothing should happen. Just return STATUS_PENDING, don't touch the Irp.

Reply:
If `CcCopyRead` would set `STATUS_PENDING`, `IoMarkIrpPending` should be called by it too, not here, Right? Removed.
I concur about `ASSERT`. Done.